@AH-64D Apache Longbow/Addons/fza_ah64_mpd/functions/tsd/fn_tsdRTEHandleControl.sqf

```diff
@@ -1,20 +1,102 @@
 params ["_heli", "_mpdIndex", "_control", "_state"];
 #include "\fza_ah64_mpd\headers\mfdConstants.h"
+#include "\fza_ah64_dms\headers\constants.h"
 #include "\fza_ah64_mpd\headers\tsd.hpp"
 
-private _variant = _state get "subPageVarPage" select 1;
+private _variant      = _state get "subPageVarPage" select 1;
+private _routeData    = _heli getVariable "fza_ah64_routeData";
+private _routeCurrent = _heli getVariable "fza_ah64_routeSelected";
+private _routeInfo    = _routeData # _routeCurrent;
+
+private _addRoutePoint = {
+    params ["_heli", "_btnIndex"];
+    private _routePoint = _heli getVariable "fza_mpd_tsdRteCurrentSel";
+    if (isNil "_routePoint") exitWith {};
+    private _rteIndex = _heli getVariable "fza_ah64_routeCurPoint";
+    private _index    = ((_state get "routeScroll") + _btnIndex);
+    private _dataType = [_heli, _routePoint, POINT_GET_TYPE] call fza_dms_fnc_pointGetValue;
+    private _dmsNext  = (_heli getVariable "fza_dms_routeNext")#0;
+    if ((count _routeInfo) < _index) exitWith {};
+    if (isNil "_dataType") exitWith {};
+    _routeInfo insert [_index, [_routePoint]];
+    _routeData set [_routeCurrent, _routeInfo];
+    if (_dmsNext isEqualTo []) then {
+        [_heli, _routePoint] call fza_dms_fnc_routeSetDir;
+        _heli setVariable ["fza_ah64_routeCurPoint", _index, true];
+    };
+    if (_index <= _rteIndex && _rteIndex != -1) then {
+        _heli setVariable ["fza_ah64_routeCurPoint", (_rteIndex + 1), true];
+    };
```

Comment:
Looking at these functions, they seem to be composed of two sections
1. Pulling the data from the MPD state
2. Performing the route operation (add, del etc.)

I would like the second part to be a separate function in the dms - should just be as easy as moving over the latter half.